### ISSUE TITLE
app not responding and account lockouts

### ISSUE THREAD

**Issue by OUTOTS** [BIG-RAT/mobile_to_local#10]
almost everytime that I've tried to use this app it freezes, goes into not responding and will stay there for 30 mins+ until I force quit. 
![image](https://github.com/BIG-RAT/mobile_to_local/assets/163584376/df963cc6-6ea2-4720-9c54-42b04f32acad)
not sure what damages if any this is doing as even when the app freezes it still migrates the account to standard/admin and unbinds just hangs indefinitely. 

And when it does work the account gives an error that it's locked out or the password used to migrate the account doesn't work to login. Have to go into a different account and change the password of the acc that was migrated.

**Comment by BIG-RAT** [comment]
What switches are used when launching the app? Done through a scheduled policy or self service? What shows up in the logs (`/private/var/log/mobile.to.local.log`)?

**Comment by OUTOTS** [comment]
I've tried a variation of switches:
sudo /Users/_username_/Downloads/Mobile\ to\ Local.app/Contents/MacOS/Mobile\ to\ Local
sudo /Users/_username_/Downloads/Mobile\ to\ Local.app/Contents/MacOS/Mobile\ to\ Local -userType standard -unbind true
sudo /Users/_username_/Downloads/Mobile\ to\ Local.app/Contents/MacOS/Mobile\ to\ Local -mode silent -userType standard

I have since reimaged the machine, will try the process again today and provide logs.

**Comment by OUTOTS** [comment]
using the silent switch "sudo /Users/username/Downloads/Mobile\ to\ Local.app/Contents/MacOS/Mobile\ to\ Local -mode silent -userType standard"

The migration was completed but the acc was then locked out after. 
[mobile.to.local.log](https://github.com/BIG-RAT/mobile_to_local/files/14669718/mobile.to.local.log)
![image](https://github.com/BIG-RAT/mobile_to_local/assets/163584376/a0e26f4e-32e6-4cc0-a0bc-fc645d369bc0)

**Comment by marcogrima93** [comment]
i am having the same thing when running with the below payload

`sudo /private/var/tmp/Mobile\ to\ Local.app/Contents/MacOS/Mobile\ to\ Local -userType standard; sleep 10; cat /private/var/log/mobile.to.local.log; sleep 5; rm -rf /private/var/tmp/Mobile\ to\ Local.app; sleep 3; rm -rf /private/var/log/mobile.to.local.log`

it goes not responding and i need to force quit, restart and log back in to the account. it seems then its a standard account but i dont know what has been broken.

also i was unbinded the device without me asking to be unbinded

please fix this

**Comment by BIG-RAT** [comment]
Can we have a look at the log to get an idea where it's stalling? Most curious if we see
`chown -R :staff /Users/<someUsername>`

**Comment by marcogrima93** [comment]
To be honest the app got hung up only once, and so far after force quitting its been ok.

i tested another few times and it worked but what always happened was the password isnt correct and i would need to go to an admin account and reset it from there.

i also modified the script with a line to reset the password but i am yet to test

`/usr/bin/dscl . passwd /Users/$currentname "$password"`

below is how the end of the script looks i always get the failed for home directory tho

```
Thu May 02 15:08:38 RAMB906 marco-sadmin[migrate]: chown -R :staff /Users/marco-sadmin
Thu May 02 15:12:30 RAMB906 marco-sadmin[migrate]: failed to updated group for home directory
Thu May 02 15:12:30 RAMB906 marco-sadmin[migrate]: marco-sadmin was added to the staff group
Thu May 02 15:12:30 RAMB906 marco-sadmin[migrate]: marco-sadmin was removed from the admin group
Thu May 02 15:12:30 RAMB906 marco-sadmin[migrate]: loginwindow restarted.
```

**Comment by BIG-RAT** [comment]
Issues have been addressed in v2.6.2
